src/components/select/select.spec.js

```diff
@@ -1486,6 +1506,27 @@ describe('<md-select>', function() {
     });
   });
 
+  // Only test with custom components on AngularJS 1.5+
+  if (angular.version.major === 1 && angular.version.minor >= 5) {
+    describe('with custom components', function() {
+
+      // The ng-ref API is new in AngularJS 1.7 and is required to reproduce the issue in this test.
+      if (angular.version.minor >= 7) {
+        it('should re-validate when the options and required value change', function() {
+          var nodes = $compile('<form name="testForm">' +
+            ' <required-form-field-component ng-ref="comp"></required-form-field-component>' +
```

Comment:
@gkalpak thank you for the very informative review! Do you know of any alternative to `ng-ref` here?

Reply:
No, no direct alternatives. You could create a simple directive to do that, but it would be overkill here imo.
The cleanest/easiest way (for the test) is probably: 

```js
var ctrl = nodes.find('required-form-field-component').controller('requiredFormFieldComponent');
```

Reply:
That works, many thanks!